torchmetrics/functional/classification/calibration_error.py

```diff
@@ -0,0 +1,123 @@
+        prop_in_bin = in_bin.float().mean()
+        if prop_in_bin.item() > 0:
+            acc_bin[i] = accuracies[in_bin].float().mean()
+            conf_bin[i] = confidences[in_bin].mean()
+            prop_bin[i] = prop_in_bin
+
+    if norm == "l1":
+        ce = torch.sum(torch.abs(acc_bin - conf_bin) * prop_bin)
+        print(ce)
+    elif norm == "max":
+        ce = torch.max(torch.abs(acc_bin - conf_bin))
+    elif norm == "l2":
+        ce = torch.sum(torch.pow(acc_bin - conf_bin, 2) * prop_bin)
+        # NOTE: debiasing is disabled in the wrapper functions. This implementation differs from that in sklearn.
+        if debias:
+            # the order here (acc_bin - 1 ) vs (1 - acc_bin) is flipped from
+            # the equation in Verified Uncertainty Prediction (Kumar et al 2019)/
+            debias_bins = (acc_bin * (acc_bin - 1) * prop_bin) / (prop_bin * accuracies.size()[0] - 1)
+            ce += torch.sum(torch.nan_to_num(debias_bins))  # replace nans with zeros if nothing appeared in a bin
+        ce = torch.sqrt(ce) if ce > 0 else torch.tensor(0)
+    else:
+        raise ValueError(f"Norm {norm} is not supported. Please select from l1, l2, or max. ")
+    return ce
+
+
+def _ce_update(preds: Tensor, target: Tensor) -> Tuple[Tensor, Tensor]:
+    _, _, mode = _input_format_classification(preds, target)
+
+    if mode == DataType.BINARY:
+        confidences, accuracies = preds, target
+    elif mode == DataType.MULTICLASS:
+        confidences, predictions = preds.max(dim=1)
+        accuracies = predictions.eq(target)
+    elif mode == DataType.MULTIDIM_MULTICLASS:
+        # reshape tensors
+        # for preds, move the class dimension to the final axis and flatten the rest
+        confidences, predictions = torch.transpose(preds, 1, -1).flatten(0, -2).max(dim=1)
+        # for targets, just flatten the target
+        accuracies = predictions.eq(target.flatten())
+    else:
+        raise ValueError(
+            f"Calibration error is not well-defined for data with size {preds.size()} and targets {target.size()}"
+        )
+
+    return confidences, accuracies
+
+
+def calibration_error(preds: Tensor, target: Tensor, n_bins: int = 15, norm: str = "l1"):
+    """
+
+        Computes the top-label calibration error as described in `https://arxiv.org/pdf/1909.10155.pdf`.
+
+        Three different norms are implemented, each corresponding to variations on the calibration error metric.
+
+        L1 norm (Expected Calibration Error)
+
+        .. math::
+            \text{Accuracy} = \frac{1}{N}\sum_i^N \|(p_i - c_i)\|
+
+        Infinity norm (Maximum Calibration Error)
+
+        .. math::
+        \text{Accuracy} =  \max_{i} (p_i - c_i)
+
+        L2 norm (Root Mean Square Calibration Error)
+
+        .. math::
+        \text{Accuracy} = \frac{1}{N}\sum_i^N (p_i - c_i)^2
+
+        Where p_i is the top-1 prediction accuracy in bin i and c_i is the average confidence of predictions in bin i.
```

Comment:
will also render this as math
```suggestion
        Where :math:`p_i` is the top-1 prediction accuracy in bin i and :math:`c_i` is the average confidence of predictions in bin i.
```